File: app/src/main/java/com/zwq65/unity/data/network/retrofit/callback/ApiException.java
package com.zwq65.unity.data.network.retrofit.callback;

/**
 * ================================================
 * api返回数据异常
 * <p>
 * Created by NIRVANA on 2018/02/05.
 * Contact with <zwq651406441@gmail.com>
 * ================================================
 */
public class ApiException extends Exception {
    public String code;
    public String message;


    public ApiException(Throwable throwable, String code) {
        super(throwable);
        this.code = code;
    }


}

File: app/src/main/java/com/zwq65/unity/data/network/retrofit/callback/ApiException.kt
package com.zwq65.unity.data.network.retrofit.callback

/**
 * ================================================
 * api返回数据异常[Exception]
 * <p>
 * Created by NIRVANA on 2018/02/05.
 * Contact with <zwq651406441@gmail.com>
 * ================================================
 */
class ApiException(throwable: Throwable, var code: String) : Exception(throwable) {
    override var message: String = ""
}
